exporter/awsxrayexporter/config.go

```diff
@@ -39,4 +39,8 @@ type Config struct {
 	ResourceARN string `mapstructure:"resource_arn"`
 	// IAM role to upload segments to a different account.
 	RoleARN string `mapstructure:"role_arn"`
+	// Optional X-Ray attribute data type: https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-annotations
+	// Defines which X-Ray data type will the OpenTelemetry span attributes be translated into.
+	// Valid values are "metadata" (default value) and "annotation".
+	AttributeDataType string `mapstructure:"attribute_data_type"`
```

Comment:
Hey - this looks close to what I was thinking, but instead of this sort of boolean, how about a `IndexedAttributes string[]`? A list of attribute keys that are set as annotation instead of metadata.

Reply:
`IndexedAttributes` sounds like a good idea. Yet if the users want to index all (or almost all, they don't care) attributes, it can be hard to configure. I'm thinking of 2 enhancements
1. Use magic word `ALL` to index all attributes. But this might conflicts with an attribute called "ALL"
2. Create another boolean option `IndexAllAttributes`

Thoughts?

Reply:
Ah yeah - let's add a boolean as well

Reply:
Revised in latest commit